compiler/optimizer/VPConstraint.cpp

```diff
@@ -1343,7 +1344,8 @@ TR::VPConstString *TR::VPConstString::create(OMR::ValuePropagation *vp, TR::Symb
 
    if (vpConstStringCriticalSection.hasVMAccess())
       {
-      void *string = *((void **) symRef->getSymbol()->castToStaticSymbol()->getStaticAddress());
+      uintptrj_t stringStaticAddr = (uintptrj_t) symRef->getSymbol()->castToStaticSymbol()->getStaticAddress();
+      void *string = (void *)TR::Compiler->om.getStaticReferenceFieldAtAddress(vp->comp(), stringStaticAddr, symRef);
```

Comment:
As far as I can tell everywhere this variable is being used it is used under a `uintptrj_t` context. Can we not change the type to `uintptrj_t` and avoid all these casts?